pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -592,6 +594,26 @@ func (r *Builder) mapNetworks(vm *model.VM, object *cnv.VirtualMachineSpec) (err
 		}
 		for _, nic := range needed {
 			networkName := fmt.Sprintf("net-%v", numNetworks)
+
+			// If the network name template is set, use it to generate the network name.
+			networkNameTemplate := r.getNetworkNameTemplate(vm)
+			if networkNameTemplate != "" {
+				// Create template data
+				templateData := api.NetworkNameTemplateData{
+					NetworkName:      mapped.Destination.Name,
+					NetworkNamespace: mapped.Destination.Namespace,
+					NetworkType:      strings.ToLower(string(mapped.Destination.Type)),
+					NetworkIndex:     numNetworks,
+				}
+
+				networkName, err = r.executeTemplate(networkNameTemplate, &templateData)
+				if err != nil {
+					// fallback to default name and reset error
```

Comment:
Maybe adding a log that the template execution failed?

Reply:
re based and added a log line